packages/vite-environment-provider-cloudflare/src/index.ts

```diff
@@ -232,68 +239,58 @@ async function createCloudflareDevEnvironment(
 
       fixedSpecifier = rawSpecifier;
 
-      try {
-        const resolveId =
-          resolveMethod === 'import' ? esmResolveId : cjsResolveId;
-        let resolvedId = await resolveId(devEnv, fixedSpecifier, referrer);
+      const resolveId =
+        resolveMethod === 'import' ? esmResolveId : cjsResolveId;
 
-        if (resolvedId.includes('?'))
-          resolvedId = resolvedId.slice(0, resolvedId.lastIndexOf('?'));
+      let resolvedId = await resolveId(devEnv, fixedSpecifier, referrer);
 
-        const redirectTo =
-          resolvedId !== rawSpecifier && resolvedId !== specifier
-            ? resolvedId
-            : undefined;
-
-        if (redirectTo) {
-          return new MiniflareResponse(null, {
-            headers: { location: resolvedId },
-            status: 301,
-          });
-        }
+      if (!resolvedId) {
+        return new MiniflareResponse(null, { status: 404 });
+      }
 
-        // and we read the code from the resolved file
-        const code: string | null = await readFile(resolvedId, 'utf8').catch(
-          () => null,
-        );
-
-        const notFound = !code;
-
-        const moduleInfo = await collectModuleInfo(code, resolvedId);
-
-        debugDumps.dumpModuleFallbackServiceLog({
-          resolveMethod,
-          referrer,
-          specifier,
-          rawSpecifier,
-          fixedSpecifier,
-          resolvedId,
-          redirectTo,
-          code,
-          isCommonJS: moduleInfo.isCommonJS,
-          notFound: notFound || undefined,
+      if (resolvedId.includes('?'))
+        resolvedId = resolvedId.slice(0, resolvedId.lastIndexOf('?'));
+
+      const redirectTo =
+        !rawSpecifier.startsWith('./') &&
+        !rawSpecifier.startsWith('../') &&
+        resolvedId !== rawSpecifier &&
+        resolvedId !== specifier
+          ? resolvedId
+          : undefined;
+
+      if (redirectTo) {
+        return new MiniflareResponse(null, {
+          headers: { location: resolvedId },
+          status: 301,
         });
+      }
 
-        if (notFound) {
-          return new MiniflareResponse(null, { status: 404 });
-        }
+      const code: string | null = await readFile(resolvedId, 'utf8');
+
+      const notFound = !code;
```

Comment:
I think this will throw rather than return `null` if the file is not found .

```suggestion
      let code: string
      
      try {
        code = await readFile(resolvedId, 'utf8');
      } catch () {
        return new MiniflareResponse(`Failed to read file ${resolvedId}`, { status: 404 });
      }
```

Reply:
yes, you're totally right thanks 🙂